### DISCUSSION TITLE
Using promises to set the default value in config does not work

### DISCUSSION THREAD

**Discussion by abhinandshibu** [Milkdown/milkdown#958] [Q&A] [answered]
I am trying to load the content which should go inside of the editor from the backend, and then set it as the default value once I have it, however it does not work. I'm wondering how to overcome this? I won't be able to use `editor.action(replaceAll("content"))` inside the config since we don't have the editor and are creating it. Ideally I would like to set the editor with the correct content from the beginning rather than loading an empty string, then setting the correct value later once it is loaded. 

``` javascript
const { get } = useEditor((root) =>
    Editor.make()
      .config(nord)
      .config((ctx) => {
        ctx.set(rootCtx, root);
        ctx.set(editorViewOptionsCtx, { editable: () => editable });
        
        loadContent().then(content => {
          console.log(content); // This console logs the correct content
          ctx.set(defaultValueCtx, content); // This does not set the default value
                   
          const listener = ctx.get(listenerCtx);
          listener.markdownUpdated((ctx, markdown, prevMarkdown) => {
            if (markdown !== prevMarkdown) {
              saveContent(markdown);
            }
          })
        });
      })  
```

**Comment by Saul-Mirone** [comment, marked as answer]
~~Operations in config should be synchronized.~~

I recheck the code and find out that the async config is also supported. But you should return a promise or use async function.

```ts
Editor.make()
  .config(async (ctx) => {
    const defaultValue = await fetchContent();
    ctx.set(defaultValueCtx, defaultValue);
  })
```

This should work.

**Reply by abhinandshibu** [reply]
Okay, thanks for letting me know

**Reply by Saul-Mirone** [reply]
I recheck the code and find out that the async config is also supported. But you should return a promise or use async function.

```ts
Editor.make()
  .config(async (ctx) => {
    const defaultValue = await fetchContent();
    ctx.set(defaultValueCtx, defaultValue);
  })
```

This should work.

**Reply by abhinandshibu** [reply]
That's great, thank you!
